src/core/Akka/Event/AddressTerminatedTopic.cs

```diff
@@ -35,7 +37,7 @@ public override AddressTerminatedTopic CreateExtension(ExtendedActorSystem syste
     /// </summary>
     internal sealed class AddressTerminatedTopic : IExtension
     {
-        private readonly AtomicReference<HashSet<IActorRef>> _subscribers = new AtomicReference<HashSet<IActorRef>>(new HashSet<IActorRef>());
```

Comment:
Another ancient bug - looks like we tried to get too cute by half here. I agree that a `lock` should be a relatively simple fix for this.